### PR TITLE
Fix parsing of $not expression on command line

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

- fixes bug
- adds test to `test_shell.py`
- I do not fix the testing code in `test_find_command_line_interface.py` that does not properly test the code path but I left a todo for now.

## Motivation and Context

The `$not` expression was broken on the command line. I noticed this while working on https://github.com/glotzerlab/signac-docs/pull/199

signac was inserting `sp.` prefixes before the operator

```bash
$ signac find '{"$not": {"a": 1}}'
Error: "Invalid operator placement 'sp.$not.a'."
```

The function `filterparse._add_prefix` does the inserting, which later triggered an error in `_search_indexer._find_expression`. This code path wasn't tested.

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac/blob/main/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac/blob/main/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/main/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac/tree/main/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/main/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: tests/test_shell.py
Comment (bdice) [inline tests/test_shell.py:285]: Can we define `FILTERS` in some common file, maybe in `conftest.py` as a fixture?

Alternatively, can we move all `import` statements to the top of the file?

File: signac/filterparse.py
Comment (bdice) [inline signac/filterparse.py:206]: Maybe add a comment above the `$and`/`$or` handling that says:

```python
# Logical operators ($and, $or, $not) should not be prefixed, but their values should.
```

File: tests/test_shell.py
Comment (cbkerr) [reply tests/test_shell.py:285]: I moved the import up, but I like having FILTERS defined near where they are most used.

Comment (bdice) [reply tests/test_shell.py]: Test files should not import from other test files. It is generally a good design principle for test files to be independent of one another.

Comment (bdice) [reply tests/test_shell.py]: Can we refactor that before merging?

**Review by bdice — APPROVED** [review]
I'd like to see `test_shell` avoid importing from `test_find_command_line_interface`, but I won't block this PR on that.

**Pull request merged**
